### ISSUE TITLE
Pylint incorrectly flags as `unused-import` things that are used in string literal type annotations

### ISSUE THREAD

**Issue by sanyassh** [PyCQA/pylint#3299]
### Steps to reproduce

pylint_example.py:

```
class Class:
    pass
```

pylint_example2.py:

```
from pylint_example import Class


class Class2:
    cls: 'Class'
```

pylint pylint_example2.py

```
************* Module pylint_example2
pylint_example2.py:4:0: R0903: Too few public methods (0/2) (too-few-public-methods)
pylint_example2.py:1:0: W0611: Unused Class imported from pylint_example (unused-import)

------------------------------------------------------------------
Your code has been rated at 3.33/10 (previous run: 3.33/10, +0.00)
```

### Current behavior

String literal in type annotation is considered "unused" and warning is emitted.

### Expected behavior

There should not be any warning about unused import.
 
### pylint --version output

pylint 2.4.4
astroid 2.3.3
Python 3.6.9 (default, Nov  7 2019, 10:44:02) 
[GCC 8.3.0]

**Comment by jnsnow** [comment]
I have some additional information related to this issue.

Even though there are workarounds by using e.g.
```python
from __future__ import annotations
```
there are some type definitions that actually still *require* you to use string quoting, because typeshed expresses some types as subscriptable generics, e.g. `os.PathLike` must be expressed as `'os.PathLike[str]'` -- the quotes are a requirement, because this type cannot be subscripted at runtime.

Sadly, every other CQA tool also fails here: pyflakes and pycharm both do not understand this annotation and all three tools have different silencing methods.

**Comment by vbraun** [comment]
Flake8 can do it for the record, and does not emit an F401 on the example:
~~~~
# flake8 pylint_example2.py
# echo $?
0
~~~~

**Comment by dmtucker** [comment]
#4315 also reproduces when the type is a string.
